tests/src/NetworkRequest.cpp

```diff
@@ -183,30 +164,21 @@ TEST(NetworkRequest, FinallyCallbackOnError)
     for (const auto code : codes)
     {
         auto url = getStatusURL(code);
-        std::mutex mut;
         bool requestDone = false;
-        std::condition_variable requestDoneCondition;
 
         bool finallyCalled = false;
 
         NetworkRequest(url)
-            .finally(
-                [&mut, &requestDone, &requestDoneCondition, &finallyCalled] {
-                    finallyCalled = true;
-
-                    {
-                        std::unique_lock lck(mut);
-                        requestDone = true;
-                    }
-                    requestDoneCondition.notify_one();
-                })
+            .finally([&requestDone, &finallyCalled] {
+                finallyCalled = true;
+                requestDone = true;
+            })
             .execute();
 
         // Wait for the request to finish
-        std::unique_lock lck(mut);
-        requestDoneCondition.wait(lck, [&requestDone] {
+        EXPECT_TRUE(QTest::qWaitFor([&requestDone] {
```

Comment:
Keep condition variable, use wait_for and manually call https://github.com/qt/qtbase/blob/dev/src/corelib/kernel/qtestsupport_core.h#L33-L40

can make helper function if it feels cleaner